src/body-builder.js

```diff
@@ -0,0 +1,197 @@
+import _ from 'lodash'
+import {boolMerge} from './utils'
+
+function bodyBuilder () {
+  return Object.assign(
+    { /* sort, from, build, etc. go here */ },
+    queryBuilder(),
+    filterBuilder(),
+    aggregationBuilder()
+  )
+}
+
+/**
+ * Generic builder for filter and query clauses
+ *
+ * @private
+ *
+ * @param  {string} field
+ * @param  {any}    value
+ * @param  {Object} opts
+ * @return {Object}       query clause component
+ */
+function buildClause (field, value, opts) {
```

Comment:
Not relevant to this MR but something on my mind is there are many variations in the DSL that aren't really captured by this `buildClause`, e.g.
- https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-ids-query.html (uses `type` and `values` instead of field)
- https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-limit-query.html (uses `value` instead of `field`)
- https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-span-multi-term-query.html (wraps a query clause)

I think ideally we'll have a test suite that covers all the variations we can expect from the DSL so that people can see examples of how to build each variation. But we can revisit this later.